app/reflexes/products_reflex.rb

```diff
@@ -28,6 +28,10 @@ def render_products
                    locals: { products: @products, pagy: @pagy })
     ).broadcast
 
+    cable_ready.replace_state(
+      url: current_url,
+    ).broadcast_later
```

Comment:
Because this change the URL, and if we change the URL before the cable_ready.replace this causes an error (cant' remember but something like : "can't update DOM since URL has changed")

Reply:
the exact error is
```
stimulus_reflex.js:690 Reflex failed due to mismatched URL.
```

Reply:
Oh.. interesting! In that case using `broadcast` (now) would probably be safer.

It seems like this should be easier to solve though. Surely this pattern is a normal one. I would think it's equivalent to using `<form method="get">`.

Reply:
I just realised that my suggestion would probably still create the error you mentioned, so I'll leave this.

Reply:
I don't really know the context. So I'm raising the obvious questions again: Why do we change the URL? Or why don't we use GET to automate that?

Reply:
> Why do we change the URL?

To save filters.

Reply:
> why don't we use GET to automate that?

For example, trigger a normal request with a form submit? 
A form submit would require Rails to rebuild the whole page, which would take slightly longer but that's not really a problem.

But we also want to implement the loading behaviour: show a loading spinner. So if we're going to intercept the basic behaviour, I think we're going to have to also change the URL with JS.

Which brings me to a conclusion: why not update the URL in `productsV3_controller.js` instead?

(caveat: I'm still getting used to all the different options we have here so might not be seeing the whole picture)

Reply:
> Which brings me to a conclusion: why not update the URL in productsV3_controller.js instead?

We don't have access to the pagination, filtering, ... values, which are handled via the reflex. Actually, I was thinking the opposite, ie. removing `app/webpacker/controllers/products_controller.js` file and only use `app/reflexes/products_reflex.rb` to have one single source of truth.

Reply:
🤔  
Hmm, maybe we should move on and reconsider this later..